chunkcache.cpp

```diff
@@ -80,9 +80,9 @@ int ChunkCache::getMaxCost() const {
 QSharedPointer<Chunk> ChunkCache::fetchCached(int cx, int cz) {
   // try to get Chunk from Cache
   ChunkID id(cx, cz);
-  mutex.lock();
+
+  QMutexLocker guard(&mutex);
   QSharedPointer<Chunk> * p_chunk(cache[id]);   // const operation
-  mutex.unlock();
 
   if (p_chunk != NULL )
     return QSharedPointer<Chunk>(*p_chunk);
```

Comment:
this copy operation here needs to be done within the mutex protected scope. otherwise the cache could have removed the shared pointer instance which p_chunk is pointing to already before.